crates/ruff_python_parser/src/lexer.rs

```diff
@@ -690,48 +690,67 @@ impl<'source> Lexer<'source> {
 
         let value_start = self.offset();
 
-        let value_end = loop {
-            match self.cursor.bump() {
-                Some('\\') => {
-                    if self.cursor.eat_char('\r') {
-                        self.cursor.eat_char('\n');
-                    } else {
-                        self.cursor.bump();
-                    }
-                }
-                Some('\r' | '\n') if !triple_quoted => {
+        let value_end = if triple_quoted {
+            // For triple-quoted strings, scan until we find the closing quote (ignoring escaped
+            // quotes) or the end of the file.
+            loop {
+                let Some(index) = memchr::memchr(quote as u8, self.cursor.rest().as_bytes()) else {
```

Comment:
Nit: I find the `let Some(...) .. else` use here slightly confusing because the `else` block is so long. That's why I would prefer a regular `if let Some(index) {...} else {...}` here

Reply:
I don't think I mind the `let ... else`, but I'd probably do `u8::try_from(quote).expect("char that fits in u8")` here instead of `quote as u8`.